ovh/data_me_api_oauth2_client.go

```diff
@@ -0,0 +1,82 @@
+package ovh
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/ovh/terraform-provider-ovh/ovh/helpers"
+)
+
+func dataSourceMeApiOauth2Client() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMeApiOauth2ClientRead,
+		Schema: map[string]*schema.Schema{
+			"callback_urls": {
+				Type:        schema.TypeList,
+				Description: "Callback URLs of the applications using this oauth2 client. Required if using the AUTHORIZATION_CODE flow.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Computed: true,
+			},
+			"client_id": {
+				Type:        schema.TypeString,
+				Description: "Client ID for the oauth2 client, generated during the resource creation.",
+				Required:    true,
+			},
+			"client_secret": {
```

Comment:
I was thinking we could use this field to specify a data source with an existing client_secret. Do you think this use case is enough?  If not I can remove it from the data source and document it better.

Reply:
I think the proper way for this would be to import a resource instead. You could declare a custom import function that would be used like this: `terraform import  ovh_me_api_oauth2_client "$id|$client_secret"` or using a variable like this: `terraform import -var "client_secret=$client_secret"  ovh_me_api_oauth2_client`.
(See doc here: https://developer.hashicorp.com/terraform/tutorials/providers-plugin-framework/providers-plugin-framework-resource-import)

Reply:
Thank you for the quick answer, I just pushed an edit for the resources to be importable with a client_secret as you suggested, and removed the field from the data source:

```bash
terraform import  ovh_me_api_oauth2_client 'client_id|client_secret'
```